In [1]:
import pandas as pd
pd.set_option('display.max_columns', None)  
import plotly.express as px
pd.options.plotting.backend = "plotly"

In [34]:
test = pd.read_csv('../data/test.csv')
test

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0     1461          20       RH         80.0    11622   Pave   NaN      Reg   
1     1462          20       RL         81.0    14267   Pave   NaN      IR1   
2     1463          60       RL         74.0    13830   Pave   NaN      IR1   
3     1464          60       RL         78.0     9978   Pave   NaN      IR1   
4     1465         120       RL         43.0     5005   Pave   NaN      IR1   
...    ...         ...      ...          ...      ...    ...   ...      ...   
1454  2915         160       RM         21.0     1936   Pave   NaN      Reg   
1455  2916         160       RM         21.0     1894   Pave   NaN      Reg   
1456  2917          20       RL        160.0    20000   Pave   NaN      Reg   
1457  2918          85       RL         62.0    10441   Pave   NaN      Reg   
1458  2919          60       RL         74.0     9627   Pave   NaN      Reg   

     LandContour Utilities LotConfig LandSlope Neighborhood Condition1  \
0            Lvl    AllPub    Inside       Gtl        NAmes      Feedr   
1            Lvl    AllPub    Corner       Gtl        NAmes       Norm   
2            Lvl    AllPub    Inside       Gtl      Gilbert       Norm   
3            Lvl    AllPub    Inside       Gtl      Gilbert       Norm   
4            HLS    AllPub    Inside       Gtl      StoneBr       Norm   
...          ...       ...       ...       ...          ...        ...   
1454         Lvl    AllPub    Inside       Gtl      MeadowV       Norm   
1455         Lvl    AllPub    Inside       Gtl      MeadowV       Norm   
1456         Lvl    AllPub    Inside       Gtl      Mitchel       Norm   
1457         Lvl    AllPub    Inside       Gtl      Mitchel       Norm   
1458         Lvl    AllPub    Inside       Mod      Mitchel       Norm   

     Condition2 BldgType HouseStyle  OverallQual  OverallCond  YearBuilt  \
0          Norm     1Fam     1Story            5            6       1961   
1          Norm     1Fam     1Story            6            6       1958   
2          Norm     1Fam     2Story            5            5       1997   
3          Norm     1Fam     2Story            6            6       1998   
4          Norm   TwnhsE     1Story            8            5       1992   
...         ...      ...        ...          ...          ...        ...   
1454       Norm    Twnhs     2Story            4            7       1970   
1455       Norm   TwnhsE     2Story            4            5       1970   
1456       Norm     1Fam     1Story            5            7       1960   
1457       Norm     1Fam     SFoyer            5            5       1992   
1458       Norm     1Fam     2Story            7            5       1993   

      YearRemodAdd RoofStyle RoofMatl Exterior1st Exterior2nd MasVnrType  \
0             1961     Gable  CompShg     VinylSd     VinylSd        NaN   
1             1958       Hip  CompShg     Wd Sdng     Wd Sdng    BrkFace   
2             1998     Gable  CompShg     VinylSd     VinylSd        NaN   
3             1998     Gable  CompShg     VinylSd     VinylSd    BrkFace   
4             1992     Gable  CompShg     HdBoard     HdBoard        NaN   
...            ...       ...      ...         ...         ...        ...   
1454          1970     Gable  CompShg     CemntBd     CmentBd        NaN   
1455          1970     Gable  CompShg     CemntBd     CmentBd        NaN   
1456          1996     Gable  CompShg     VinylSd     VinylSd        NaN   
1457          1992     Gable  CompShg     HdBoard     Wd Shng        NaN   
1458          1994     Gable  CompShg     HdBoard     HdBoard    BrkFace   

      MasVnrArea ExterQual ExterCond Foundation BsmtQual BsmtCond  \
0            0.0        TA        TA     CBlock       TA       TA   
1          108.0        TA        TA     CBlock       TA       TA   
2            0.0        TA        TA      PConc       Gd       TA   
3           20.0        TA        TA      PConc       TA       TA   

In [35]:
train = pd.read_csv('../data/train.csv')
train

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0        1          60       RL         65.0     8450   Pave   NaN      Reg   
1        2          20       RL         80.0     9600   Pave   NaN      Reg   
2        3          60       RL         68.0    11250   Pave   NaN      IR1   
3        4          70       RL         60.0     9550   Pave   NaN      IR1   
4        5          60       RL         84.0    14260   Pave   NaN      IR1   
...    ...         ...      ...          ...      ...    ...   ...      ...   
1455  1456          60       RL         62.0     7917   Pave   NaN      Reg   
1456  1457          20       RL         85.0    13175   Pave   NaN      Reg   
1457  1458          70       RL         66.0     9042   Pave   NaN      Reg   
1458  1459          20       RL         68.0     9717   Pave   NaN      Reg   
1459  1460          20       RL         75.0     9937   Pave   NaN      Reg   

     LandContour Utilities LotConfig LandSlope Neighborhood Condition1  \
0            Lvl    AllPub    Inside       Gtl      CollgCr       Norm   
1            Lvl    AllPub       FR2       Gtl      Veenker      Feedr   
2            Lvl    AllPub    Inside       Gtl      CollgCr       Norm   
3            Lvl    AllPub    Corner       Gtl      Crawfor       Norm   
4            Lvl    AllPub       FR2       Gtl      NoRidge       Norm   
...          ...       ...       ...       ...          ...        ...   
1455         Lvl    AllPub    Inside       Gtl      Gilbert       Norm   
1456         Lvl    AllPub    Inside       Gtl       NWAmes       Norm   
1457         Lvl    AllPub    Inside       Gtl      Crawfor       Norm   
1458         Lvl    AllPub    Inside       Gtl        NAmes       Norm   
1459         Lvl    AllPub    Inside       Gtl      Edwards       Norm   

     Condition2 BldgType HouseStyle  OverallQual  OverallCond  YearBuilt  \
0          Norm     1Fam     2Story            7            5       2003   
1          Norm     1Fam     1Story            6            8       1976   
2          Norm     1Fam     2Story            7            5       2001   
3          Norm     1Fam     2Story            7            5       1915   
4          Norm     1Fam     2Story            8            5       2000   
...         ...      ...        ...          ...          ...        ...   
1455       Norm     1Fam     2Story            6            5       1999   
1456       Norm     1Fam     1Story            6            6       1978   
1457       Norm     1Fam     2Story            7            9       1941   
1458       Norm     1Fam     1Story            5            6       1950   
1459       Norm     1Fam     1Story            5            6       1965   

      YearRemodAdd RoofStyle RoofMatl Exterior1st Exterior2nd MasVnrType  \
0             2003     Gable  CompShg     VinylSd     VinylSd    BrkFace   
1             1976     Gable  CompShg     MetalSd     MetalSd        NaN   
2             2002     Gable  CompShg     VinylSd     VinylSd    BrkFace   
3             1970     Gable  CompShg     Wd Sdng     Wd Shng        NaN   
4             2000     Gable  CompShg     VinylSd     VinylSd    BrkFace   
...            ...       ...      ...         ...         ...        ...   
1455          2000     Gable  CompShg     VinylSd     VinylSd        NaN   
1456          1988     Gable  CompShg     Plywood     Plywood      Stone   
1457          2006     Gable  CompShg     CemntBd     CmentBd        NaN   
1458          1996       Hip  CompShg     MetalSd     MetalSd        NaN   
1459          1965     Gable  CompShg     HdBoard     HdBoard        NaN   

      MasVnrArea ExterQual ExterCond Foundation BsmtQual BsmtCond  \
0          196.0        Gd        TA      PConc       Gd       TA   
1            0.0        TA        TA     CBlock       Gd       TA   
2          162.0        Gd        TA      PConc       Gd       TA   
3            0.0        TA        TA     BrkTil       TA       Gd   

Debemos hacer las mismas transformaciones en ambos datasets, por lo que los concatenaremos para aplicar en ambos a la vez las trasnformaciones que convengamos. Una vez terminado los volveremos a separar.

In [36]:
data = pd.concat([train.drop(columns='SalePrice'),test],axis=0).reset_index(drop=True)
data

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0        1          60       RL         65.0     8450   Pave   NaN      Reg   
1        2          20       RL         80.0     9600   Pave   NaN      Reg   
2        3          60       RL         68.0    11250   Pave   NaN      IR1   
3        4          70       RL         60.0     9550   Pave   NaN      IR1   
4        5          60       RL         84.0    14260   Pave   NaN      IR1   
...    ...         ...      ...          ...      ...    ...   ...      ...   
2914  2915         160       RM         21.0     1936   Pave   NaN      Reg   
2915  2916         160       RM         21.0     1894   Pave   NaN      Reg   
2916  2917          20       RL        160.0    20000   Pave   NaN      Reg   
2917  2918          85       RL         62.0    10441   Pave   NaN      Reg   
2918  2919          60       RL         74.0     9627   Pave   NaN      Reg   

     LandContour Utilities LotConfig LandSlope Neighborhood Condition1  \
0            Lvl    AllPub    Inside       Gtl      CollgCr       Norm   
1            Lvl    AllPub       FR2       Gtl      Veenker      Feedr   
2            Lvl    AllPub    Inside       Gtl      CollgCr       Norm   
3            Lvl    AllPub    Corner       Gtl      Crawfor       Norm   
4            Lvl    AllPub       FR2       Gtl      NoRidge       Norm   
...          ...       ...       ...       ...          ...        ...   
2914         Lvl    AllPub    Inside       Gtl      MeadowV       Norm   
2915         Lvl    AllPub    Inside       Gtl      MeadowV       Norm   
2916         Lvl    AllPub    Inside       Gtl      Mitchel       Norm   
2917         Lvl    AllPub    Inside       Gtl      Mitchel       Norm   
2918         Lvl    AllPub    Inside       Mod      Mitchel       Norm   

     Condition2 BldgType HouseStyle  OverallQual  OverallCond  YearBuilt  \
0          Norm     1Fam     2Story            7            5       2003   
1          Norm     1Fam     1Story            6            8       1976   
2          Norm     1Fam     2Story            7            5       2001   
3          Norm     1Fam     2Story            7            5       1915   
4          Norm     1Fam     2Story            8            5       2000   
...         ...      ...        ...          ...          ...        ...   
2914       Norm    Twnhs     2Story            4            7       1970   
2915       Norm   TwnhsE     2Story            4            5       1970   
2916       Norm     1Fam     1Story            5            7       1960   
2917       Norm     1Fam     SFoyer            5            5       1992   
2918       Norm     1Fam     2Story            7            5       1993   

      YearRemodAdd RoofStyle RoofMatl Exterior1st Exterior2nd MasVnrType  \
0             2003     Gable  CompShg     VinylSd     VinylSd    BrkFace   
1             1976     Gable  CompShg     MetalSd     MetalSd        NaN   
2             2002     Gable  CompShg     VinylSd     VinylSd    BrkFace   
3             1970     Gable  CompShg     Wd Sdng     Wd Shng        NaN   
4             2000     Gable  CompShg     VinylSd     VinylSd    BrkFace   
...            ...       ...      ...         ...         ...        ...   
2914          1970     Gable  CompShg     CemntBd     CmentBd        NaN   
2915          1970     Gable  CompShg     CemntBd     CmentBd        NaN   
2916          1996     Gable  CompShg     VinylSd     VinylSd        NaN   
2917          1992     Gable  CompShg     HdBoard     Wd Shng        NaN   
2918          1994     Gable  CompShg     HdBoard     HdBoard    BrkFace   

      MasVnrArea ExterQual ExterCond Foundation BsmtQual BsmtCond  \
0          196.0        Gd        TA      PConc       Gd       TA   
1            0.0        TA        TA     CBlock       Gd       TA   
2          162.0        Gd        TA      PConc       Gd       TA   
3            0.0        TA        TA     BrkTil       TA       Gd   

In [37]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2919 entries, 0 to 2918
Data columns (total 80 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             2919 non-null   int64  
 1   MSSubClass     2919 non-null   int64  
 2   MSZoning       2915 non-null   object 
 3   LotFrontage    2433 non-null   float64
 4   LotArea        2919 non-null   int64  
 5   Street         2919 non-null   object 
 6   Alley          198 non-null    object 
 7   LotShape       2919 non-null   object 
 8   LandContour    2919 non-null   object 
 9   Utilities      2917 non-null   object 
 10  LotConfig      2919 non-null   object 
 11  LandSlope      2919 non-null   object 
 12  Neighborhood   2919 non-null   object 
 13  Condition1     2919 non-null   object 
 14  Condition2     2919 non-null   object 
 15  BldgType       2919 non-null   object 
 16  HouseStyle     2919 non-null   object 
 17  OverallQual    2919 non-null   int64  
 18  OverallC

Hay muchos nulos. Vamos a visualizarlos

In [38]:
data.isna().sum()[data.isna().sum() > 0].plot.bar()

## Revisión de nulos
1. Alley

In [39]:
data.Alley.unique()

array([nan, 'Grvl', 'Pave'], dtype=object)

In [40]:
data.Alley.value_counts()

Alley
Grvl    120
Pave     78
Name: count, dtype: int64

Según la inforamción que acompaña al dataset los valores nulos, son propiedades que no tienen Alley

In [41]:
data.loc[data.Alley.isna(),'Alley'] = 'No Alley'

In [42]:
data.Alley.value_counts()

Alley
No Alley    2721
Grvl         120
Pave          78
Name: count, dtype: int64

2. PoolQC

In [43]:
data.PoolQC.unique()

array([nan, 'Ex', 'Fa', 'Gd'], dtype=object)

In [44]:
data.PoolQC.value_counts()

PoolQC
Ex    4
Gd    4
Fa    2
Name: count, dtype: int64

Según documentación los valores no nulos son sin piscina. El número de nulos es elevado de casi 3000 valores son solo 10 no nulos. Teniendo 80 campos que analizar, optaremos por eliminar esta columna, ya que demasiadas características puede aplicar mucho ruido al modelo.

In [45]:
data = data.drop(columns='PoolQC')

3. MiscFeature

In [46]:
data.MiscFeature.unique()

array([nan, 'Shed', 'Gar2', 'Othr', 'TenC'], dtype=object)

In [47]:
data.MiscFeature.info()

<class 'pandas.core.series.Series'>
RangeIndex: 2919 entries, 0 to 2918
Series name: MiscFeature
Non-Null Count  Dtype 
--------------  ----- 
105 non-null    object
dtypes: object(1)
memory usage: 22.9+ KB


In [48]:
data.MiscFeature.value_counts()

MiscFeature
Shed    95
Gar2     5
Othr     4
TenC     1
Name: count, dtype: int64

Lo catalogaremos como una categoría de 1 y 0. Si tiene MiscFeature = 1 si no 0. Lo pasamos a int, ya que tenga una cualidad extra la casa tendrá más valor

In [49]:
data.loc[~(data.MiscFeature.isna()),'MiscFeature'] = 1

In [50]:
data.loc[data.MiscFeature.isna(),'MiscFeature'] = 0

In [53]:
data.MiscFeature = data.MiscFeature.astype('int8')

In [54]:
data.MiscFeature.value_counts()

MiscFeature
0    2814
1     105
Name: count, dtype: int64

4. Fence

In [55]:
data.Fence.unique()

array([nan, 'MnPrv', 'GdWo', 'GdPrv', 'MnWw'], dtype=object)

In [56]:
data.Fence.info()

<class 'pandas.core.series.Series'>
RangeIndex: 2919 entries, 0 to 2918
Series name: Fence
Non-Null Count  Dtype 
--------------  ----- 
571 non-null    object
dtypes: object(1)
memory usage: 22.9+ KB


In [57]:
data.Fence.value_counts()

Fence
MnPrv    329
GdPrv    118
GdWo     112
MnWw      12
Name: count, dtype: int64

Actuamos de manera similar a la anterior, y calificamos con 1 si tiene Fence y 0 si no tiene. No podemos distinguir sin good wood es mejor que good privicy. 

In [58]:
data.loc[~(data.Fence.isna()),'Fence'] = 1
data.loc[data.Fence.isna(),'Fence'] = 0
data.Fence = data.Fence.astype('int8')

In [59]:
data.Fence.value_counts()

Fence
0    2348
1     571
Name: count, dtype: int64

5. MassVnrType

Los valores nulos según documentación significa que no tienen revistimiento

In [60]:
data.MasVnrType = data.MasVnrType.fillna('No mass')

In [61]:
data.MasVnrType.value_counts()

MasVnrType
No mass    1766
BrkFace     879
Stone       249
BrkCmn       25
Name: count, dtype: int64

6. FireplaceQu

Según documentación, los valores nulos son que no posee fireplace

In [62]:
data.FireplaceQu = data.FireplaceQu.fillna(0)

In [67]:
data.FireplaceQu = data.FireplaceQu.replace({'Ex':5,'Gd':4,'TA':3,'Fa':2,'Po':1})

In [68]:
data.FireplaceQu = data.FireplaceQu.astype('int8')

7. LotFrontage

Los valores nulos son aquellas viviendas que no tienen comunicación directa con la calle (pisos). Por lo tanto los valores nulos los podemos igualar a cero

In [71]:
data.LotFrontage = data.LotFrontage.fillna(0)

8. Garage Columns

Vemos que hay muchas columnas realacionadas con el garage que tienen la misma cantidad de nulos (159). Revisando la documentación vemos que los valores nulos son aquellos que no tienen garaje. Por lo que los rellenaremos con esta información. Revisaremos también aquellas columnas categóricas que nos interese pasar a numérica, con le objetivo de mejorar la regresión

In [73]:
# GarageType. La mantenemos categórica
data.GarageType = data.GarageType.fillna('NoGar')

In [78]:
# GarageYrBlt. Año de construcción del garaje
data.GarageYrBlt.describe()

count    2760.000000
mean     1978.113406
std        25.574285
min      1895.000000
25%      1960.000000
50%      1979.000000
75%      2002.000000
max      2207.000000
Name: GarageYrBlt, dtype: float64

Para rellenar esta columna optaremos por establecer este valor como un outlier. No es conveniente rellenarlos con 0 porque es un número bastante menor que la media, y puede afectar negativamente al rendimiento del modelo. Así que las casas que no tienen garaje, estableceremos el valor de año de construcción (ficiticio) como media-3sigma

In [80]:
data.GarageYrBlt = data.GarageYrBlt.fillna(data.GarageYrBlt.describe()['mean']-3*data.GarageYrBlt.describe()['std'])

In [83]:
data.GarageYrBlt.sort_values(ascending=False)

2592    2207.0
378     2010.0
819     2010.0
987     2010.0
1608    2010.0
         ...  
2250    1900.0
2757    1900.0
1530    1900.0
1602    1896.0
2217    1895.0
Name: GarageYrBlt, Length: 2919, dtype: float64

In [91]:
data.iloc[2590:2593,:]

Id  MSSubClass MSZoning  LotFrontage  LotArea Street     Alley  \
2590  2591          20       RL        149.0    19958   Pave  No Alley   
2591  2592          20       RL         67.0     8368   Pave  No Alley   
2592  2593          20       RL         68.0     8298   Pave  No Alley   

     LotShape LandContour Utilities LotConfig LandSlope Neighborhood  \
2590      Reg         Lvl    AllPub    Inside       Gtl       Timber   
2591      IR1         HLS    AllPub    Inside       Gtl       Timber   
2592      IR1         HLS    AllPub    Inside       Gtl       Timber   

     Condition1 Condition2 BldgType HouseStyle  OverallQual  OverallCond  \
2590       Norm       Norm     1Fam     1Story            6            8   
2591       Norm       Norm     1Fam     1Story            7            5   
2592       Norm       Norm     1Fam     1Story            8            5   

      YearBuilt  YearRemodAdd RoofStyle RoofMatl Exterior1st Exterior2nd  \
2590       1958          1995       Hip  CompShg     HdBoard     HdBoard   
2591       2006          2007     Gable  CompShg     VinylSd     VinylSd   
2592       2006          2007       Hip  CompShg     VinylSd     VinylSd   

     MasVnrType  MasVnrArea ExterQual ExterCond Foundation BsmtQual BsmtCond  \
2590    BrkFace      1224.0        TA        Gd     CBlock       TA       TA   
2591    No mass         0.0        Gd        TA      PConc       Gd       TA   
2592    No mass         NaN        Gd        TA      PConc       Gd       TA   

     BsmtExposure BsmtFinType1  BsmtFinSF1 BsmtFinType2  BsmtFinSF2  \
2590           No          Unf         0.0          Unf         0.0   
2591           No          Unf         0.0          Unf         0.0   
2592           Av          GLQ       583.0          Unf         0.0   

      BsmtUnfSF  TotalBsmtSF Heating HeatingQC CentralAir Electrical  \
2590      585.0        585.0    GasA        Gd          Y      SBrkr   
2591     1689.0       1689.0    GasA        Ex          Y      SBrkr   
2592      963.0       1546.0    GasA        Ex          Y      SBrkr   

      1stFlrSF  2ndFlrSF  LowQualFinSF  GrLivArea  BsmtFullBath  BsmtHalfBath  \
2590      2279         0             0       2279           0.0           0.0   
2591      1689         0             0       1689           0.0           0.0   
2592      1564         0             0       1564           0.0           0.0   

      FullBath  HalfBath  BedroomAbvGr  KitchenAbvGr KitchenQual  \
2590         2         1             4             1          Gd   
2591         2         0             3             1          Gd   
2592         2         0             2             1          Ex   

      TotRmsAbvGrd Functional  Fireplaces  FireplaceQu GarageType  \
2590             7        Typ           1            4     Attchd   
2591             7        Typ           0            0     Attchd   
2592             6        Typ           1            4     Attchd   

      GarageYrBlt GarageFinish  GarageCars  GarageArea GarageQual GarageCond  \
2590       1958.0          RFn         2.0       461.0         TA         TA   
2591       2007.0          Fin         2.0       433.0         TA         TA   
2592       2207.0          RFn         2.0       502.0         TA         TA   

     PavedDrive  WoodDeckSF  OpenPorchSF  EnclosedPorch  3SsnPorch  \
2590          Y         274            0              0          0   
2591          Y         100           39              0          0   
2592          Y         132            0              0          0   

      ScreenPorch  PoolArea  Fence  MiscFeature  MiscVal  MoSold  YrSold  \
2590          138         0      1            0        0       7    2007   
2591            0         0      0            0        0       4    2007   
2592            0         0      0            0        0       9    2007   

     SaleType SaleCondition  
2590       WD        Normal  
2591      New       Partial  
2592      New       Partial

Observamos que el indice 2592 tiene un valor de año de construcción del garaje de 2207. Obviemente es un error. Observamos que el año de construcción de la casa es de 2007, por lo que sustituiremos dicho valor incorrecto por 2007

In [92]:
data.loc[2592,'GarageYrBlt'] = 2007

In [93]:
data.GarageYrBlt.describe()

count    2919.000000
mean     1973.865741
std        30.057196
min      1895.000000
25%      1957.000000
50%      1977.000000
75%      2001.000000
max      2010.000000
Name: GarageYrBlt, dtype: float64

In [95]:
# GarageFinish
data.GarageFinish.unique()

array(['RFn', 'Unf', 'Fin', nan], dtype=object)

Podemos pasarlo a numérico

In [97]:
data.GarageFinish = data.GarageFinish.fillna(0)
data.GarageFinish.replace({'Fin':3,'RFn':2,'Unf':1},inplace=True)
data.GarageFinish = data.GarageFinish.astype('int8')

In [101]:
# GarageQual
data.GarageQual.unique()

array(['TA', 'Fa', 'Gd', nan, 'Ex', 'Po'], dtype=object)

Procedemos de igual manera

In [102]:
data.GarageQual = data.GarageQual.fillna(0)
data.GarageQual.replace({'Ex':5,'Gd':4,'TA':3,'Fa':2,'Po':1},inplace=True)
data.GarageQual = data.GarageQual.astype('int8')

In [ ]:
# GarageCond

In [103]:
data.GarageCond.unique()

array(['TA', 'Fa', nan, 'Gd', 'Po', 'Ex'], dtype=object)

Idem

In [104]:
data.GarageCond = data.GarageCond.fillna(0)
data.GarageCond.replace({'Ex':5,'Gd':4,'TA':3,'Fa':2,'Po':1},inplace=True)
data.GarageCond = data.GarageCond.astype('int8')

9. Basement columns

Hay que revisar los valores de estas columnas, ya que según la documentación los nulos son casas que no tienen sótano. Pero hay algunas columnas que tienen 81 nulos y otras 79, por lo que alguna información puede estar incorrecta

In [108]:
data[(data.BsmtCond.isna()) & (data.BsmtQual.notna())][['BsmtCond','BsmtQual','BsmtFinType1','TotalBsmtSF']]

BsmtCond BsmtQual BsmtFinType1  TotalBsmtSF
2040      NaN       Gd          GLQ       1426.0
2185      NaN       TA          BLQ       1127.0
2524      NaN       TA          ALQ        995.0

### Primeras impresiones:
- <ins>Id</ins>: Borraremos esta columna es solo identificativa de la línea
- <ins>MSSubClass</ins>: Esta columna está clasificada como int, y define la categoría del edificio. Se deberá pasar a string o otro tipo para que el valor numérico no afecte a la regresión